src/main/java/seedu/duke/component/Component.java

```diff
@@ -24,7 +26,9 @@ public double getValue() {
      * @param value Value of Component.
      */
     public void setValue(double value) {
-        this.value = value;
+        if (value > 0) {
+            this.value = value;
+        }
```

Comment:
Could this `throw` a `DukeException` then?

Reply:
Noted, thanks!